Cannot Inject Configuration Interface into ContainerRequestFilter Through Constructor

**Describe the bug**

Injecting configuration interface into `ContainerRequestFilter` through constructor-based injection creates 500 respones with following error message:

```
org.jboss.resteasy.spi.UnhandledException: java.lang.NullPointerException
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:381)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:216)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:168)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:245)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:122)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:36)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:87)
	at io.quarkus.runtime.CleanableExecutor$CleaningRunnable.run(CleanableExecutor.java:231)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2027)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1551)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1442)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at java.base/java.lang.Thread.run(Thread.java:830)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: java.lang.NullPointerException
	at org.jboss.resteasy.core.ConstructorInjectorImpl.<init>(ConstructorInjectorImpl.java:54)
	at org.jboss.resteasy.core.InjectorFactoryImpl.createConstructor(InjectorFactoryImpl.java:61)
	at io.quarkus.resteasy.common.runtime.QuarkusInjectorFactory.createConstructor(QuarkusInjectorFactory.java:33)
	at org.jboss.resteasy.core.providerfactory.ResteasyProviderFactoryImpl.injectedInstance(ResteasyProviderFactoryImpl.java:1397)
	at org.jboss.resteasy.core.interception.jaxrs.JaxrsInterceptorRegistryImpl$AbstractInterceptorFactory.createInterceptor(JaxrsInterceptorRegistryImpl.java:150)
	at org.jboss.resteasy.core.interception.jaxrs.JaxrsInterceptorRegistryImpl$OnDemandInterceptorFactory.initialize(JaxrsInterceptorRegistryImpl.java:168)
	at org.jboss.resteasy.core.interception.jaxrs.JaxrsInterceptorRegistryImpl$OnDemandInterceptorFactory.checkInitialize(JaxrsInterceptorRegistryImpl.java:183)
	at org.jboss.resteasy.core.interception.jaxrs.JaxrsInterceptorRegistryImpl$OnDemandInterceptorFactory.getInterceptor(JaxrsInterceptorRegistryImpl.java:194)
	at org.jboss.resteasy.core.interception.jaxrs.JaxrsInterceptorRegistryImpl$AbstractInterceptorFactory.preMatch(JaxrsInterceptorRegistryImpl.java:100)
	at org.jboss.resteasy.core.interception.jaxrs.JaxrsInterceptorRegistryImpl.preMatch(JaxrsInterceptorRegistryImpl.java:263)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
```

**Expected behavior**

`ContainerRequestFilter` should work with construtor-based injection to avoid warnings:

> Found unrecommended usage of private members (use package-private instead) in application beans:

However, if I add `@ApplicationScoped` to the filter it works. 

It would be nice if quarkus provides some diagnose.

**To Reproduce**
Steps to reproduce the behavior:
1. I have a configuration interface (as described here):
```java
@ConfigProperties
public interface SomeConfiguration {}
```
2. And I want to inject it into a `ContainerRequestFilter`:
```java
@Provider
@PreMatching
public class SomeFilter implements ContainerRequestFilter {

    private final SomeConfiguration configuration;

    @Inject
    public SomeFilter( SomeConfiguration configuration ) {
        this.configuration = configuration;
    }

}
``` 
3. Run the application and make a request


**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:  Darwin Kernel Version 19.0.0: Thu Oct 17 16:17:15 PDT 2019; root:xnu-6153.41.3~29/RELEASE_X86_64 x86_64
 - Output of `java -version`: openjdk version "13.0.2" 2020-01-14
 - Quarkus version or git rev: 1.3.0
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): mvn 3.6.1

--- a/extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java
+++ b/extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java
@@ -23,6 +23,7 @@ import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.AnnotationTarget;
 import org.jboss.jandex.AnnotationValue;
 import org.jboss.jandex.AnnotationValue.Kind;
+import org.jboss.jandex.ClassInfo;
 import org.jboss.jandex.DotName;
 import org.jboss.jandex.FieldInfo;
 import org.jboss.jandex.IndexView;
@@ -146,6 +147,7 @@ public class ResteasyCommonProcessor {
                 contributedProviders.add(i.target().asClass().name().toString());
             }
             checkProperConfigAccessInProvider(i);
+            checkProperConstructorInProvider(i);
         }
 
         Set<String> availableProviders = ServiceUtil.classNamesNamedIn(getClass().getClassLoader(),
@@ -278,6 +280,16 @@ public class ResteasyCommonProcessor {
                             + " into a JAX-RS provider may lead to unexpected results. To ensure proper results, please change the type of the field to "
                             + ParameterizedType.create(ResteasyDotNames.CDI_INSTANCE, new Type[] { fieldType }, null)
                             + ". Offending field is '" + field.name() + "' of class '" + field.declaringClass() + "'");
+        }
+    }
+
+    private void checkProperConstructorInProvider(AnnotationInstance i) {
+        ClassInfo targetClass = i.target().asClass();
+        if (!targetClass.hasNoArgsConstructor()
+                || targetClass.methods().stream().filter(m -> m.name().equals("<init>")).count() > 1) {
+            LOGGER.warn(
+                    "Classes annotated with @Provider should have a single, no-argument constructor, otherwise dependency injection won't work properly. Offending class is "
+                            + targetClass);
         }
     }
 

